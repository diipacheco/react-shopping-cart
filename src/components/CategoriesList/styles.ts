import styled from 'styled-components';

export const Container = styled.ul`
  display: flex;
  justify-content: center;

  padding: 20px 0;

  > li {
    list-style: none;
    color: var(--dark-orange);
    font-size: 14px;
    font-weight: 700;
  }
`;
