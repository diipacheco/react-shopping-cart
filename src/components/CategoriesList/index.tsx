import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

import { useProducts } from '../../context/Products';

import { Container } from './styles';

export default function CategoriesList() {
  const { categories, fetchCategories } = useProducts();

  useEffect(() => {
    fetchCategories();
  }, [fetchCategories]);
  return (
    <Container data-testid="categories-list">
      <nav>
        {categories?.map(category => (
          <li data-testid="content-categories-list" key={category}>
            <Link to={`/${category}`}>{category}</Link>
          </li>
        ))}
      </nav>
    </Container>
  );
}
