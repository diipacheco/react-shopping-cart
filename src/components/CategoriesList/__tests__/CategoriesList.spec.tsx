import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import CategoriesList from '..';

jest.mock('../../../context/Products', () => ({
  useProducts: () => ({
    categories: [
      'electronics',
      'jewelery',
      "men's clothing",
      "women's clothing",
    ],
    products: [],
    fetchCategories: jest.fn(),
  }),
}));

describe('<CategoriesList />', () => {
  it('should match snapshots', () => {
    const wrapper = render(
      <BrowserRouter>
        <CategoriesList />
      </BrowserRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to list categories', async () => {
    const { queryByTestId, findAllByTestId } = render(<CategoriesList />);

    const categoriesList = queryByTestId('categories-list') as HTMLUListElement;

    const contentCategoriesList = (await findAllByTestId(
      'content-categories-list',
    )) as HTMLLIElement[];

    expect(categoriesList.children).toHaveLength(4);
    expect(contentCategoriesList[0].innerHTML).toBe('electronics');
  });
});
