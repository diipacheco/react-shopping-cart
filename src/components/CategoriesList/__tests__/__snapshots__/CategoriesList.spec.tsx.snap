// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoriesList /> should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-bczRLJ iqorin"
        data-testid="categories-list"
      >
        <nav>
          <li
            data-testid="content-categories-list"
          >
            <a
              href="/electronics"
            >
              electronics
            </a>
          </li>
          <li
            data-testid="content-categories-list"
          >
            <a
              href="/jewelery"
            >
              jewelery
            </a>
          </li>
          <li
            data-testid="content-categories-list"
          >
            <a
              href="/men's clothing"
            >
              men's clothing
            </a>
          </li>
          <li
            data-testid="content-categories-list"
          >
            <a
              href="/women's clothing"
            >
              women's clothing
            </a>
          </li>
        </nav>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-bczRLJ iqorin"
      data-testid="categories-list"
    >
      <nav>
        <li
          data-testid="content-categories-list"
        >
          <a
            href="/electronics"
          >
            electronics
          </a>
        </li>
        <li
          data-testid="content-categories-list"
        >
          <a
            href="/jewelery"
          >
            jewelery
          </a>
        </li>
        <li
          data-testid="content-categories-list"
        >
          <a
            href="/men's clothing"
          >
            men's clothing
          </a>
        </li>
        <li
          data-testid="content-categories-list"
        >
          <a
            href="/women's clothing"
          >
            women's clothing
          </a>
        </li>
      </nav>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
